src/site/content/en/blog/asyncify/index.md

```diff
@@ -0,0 +1,514 @@
+synchronously, and both are the most limiting options in what you can store and for how long. All
+the other options provide only asynchronous APIs.
+
+This is one of the core properties of executing code on the web: any time-consuming operation, which
+includes any I/O, has to be asynchronous.
+
+The reason is that the web is historically single-threaded, and any user code that touches the UI,
+has to run on the same thread as the UI. It has to compete with the other important tasks like
+layout, rendering and event handling for the CPU time. You wouldn't want a piece of JavaScript or
+WebAssembly to be able to start a "file read" operation and block everything else—the entire tab,
+or, in the past, the entire browser—for a range from milliseconds to a few seconds, until it's over.
+
+Instead, code is only allowed to schedule an I/O operation together with a callback to be executed
+once it's finished. Such callbacks are executed as part of the browser's event loop. I won't be
+going into details here, but if you're interested in learning how event loop works under the hood,
+check out Jake Archibald's
+[article](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) which explains this
+topic in-depth.
+
+The short version is that the browser runs all the pieces of code in sort of an infinite loop, by
+taking them from the queue one by one. When some event is triggered, the browser queues the
+corresponding handler, and on the next loop iteration it's taken out from the queue and executed.
+This mechanism allows to simulate concurrency and run lots of parallel operations while still using
+a single thread.
+
+The important thing to remember about this mechanism is that, while your custom JavaScript (or
+WebAssembly) code executes, the event loop is blocked and, while it is, there is no way to react to
+any external handlers, events, I/O, etc. The only way to get the I/O results back is to register a
+callback, finish executing your code, and give the control back to the browser so that it can keep
+processing any pending tasks. Once I/O is finished, your handler will become one of those tasks and
+will get executed.
+
+For example, if you wanted to rewrite the samples above in modern JavaScript and decided to read a
+name from a remote URL, you would use Fetch API and async-await syntax:
+
+```js
+async function main() {
+  let response = await fetch("name.txt");
+  let name = await response.text();
+  console.log("Hello, %s!", name);
+}
+```
+
+Even though it looks synchronous, under the hood each `await` is essentially syntax sugar for
+callbacks:
+
+```js
+function main() {
+  return fetch("name.txt")
+    .then(response => response.text())
+    .then(name => console.log("Hello, %s!", name));
+}
+```
+
+From the desugared example, it's a bit clearer that it starts a request and subscribes to the
+response with the first callback. Once the browser receives the initial response—just the HTTP
```

Comment:
```suggestion
In this example, which is a bit clearer, a request is started and responses are subscribed to with the first callback. Once the browser receives the initial response—just the HTTP
```

Reply:
I'd like to keep "de-sugared" here to reiterate that it's not a new example, but a de-sugared version of the previous one. How does this sound?

```suggestion
In this de-sugared example, which is a bit clearer, a request is started and responses are subscribed to with the first callback. Once the browser receives the initial response—just the HTTP
```

Reply:
Your rewrite looks good.